rapids_build_backend/impls.py

```diff
@@ -8,12 +8,19 @@
 from functools import lru_cache
 from importlib import import_module
 
+import rapids_dependency_file_generator
 import tomli_w
-from packaging.requirements import Requirement
-from packaging.specifiers import SpecifierSet
 
+from . import utils
 from .config import Config
-from .utils import _get_pyproject
+
+
+def _parse_matrix(matrix):
+    if not matrix:
+        return None
+    return {
+        key: [value] for key, value in (item.split("=") for item in matrix.split(";"))
```

Comment:
This fails with:

```
    def _parse_matrix(matrix):
        if not matrix:
            return None
        return {
>           key: [value] for key, value in item.split("=") for item in matrix.split(";")
        }
E       NameError: name 'item' is not defined
```

Reply:
Oh interesting, there are a couple of things here:
1. The order needs to be flipped due to the weird scoping rules in comprehensions. `key: [value] for item in matrix.split(";") for key, value in item.split("=")`
2. This still won't work, though, because the output of `item.split('=')` isn't actually something that you want iterate over. By adding the parenthesis that you have, you're creating an iterable with one item where that first item is the key-value pair. 
3. You could write `key for item in matrix.split(";") for key in item.split("=")` and that would parse, but then you can't make a dict, only a flat list.

A more faithful representation of what's really happening here rather than the nested genex to artificially produce a second iterable would probably be something like
```
>>> matrix = "x=1;y=2;z=3"
>>> {key: value for key, value in map(lambda s: s.split('='), matrix.split(";"))}
{'x': '1', 'y': '2', 'z': '3'}
```

I'm fine leaving it too though, mostly doing the analysis to understand what's happening. Up to you which you prefer.

Reply:
This is also equivalent:

```
>>> matrix = "x=1;y=2;z=3"
>>> dict(map(lambda s: s.split("="), matrix.split(";")))
{'x': '1', 'y': '2', 'z': '3'}
```

Reply:
I know that you typed out that output instead of copy-pasting it because one of the brackets is backwards :joy: 

But yes good call, I forgot about the extra wrapping in a list. I'm fine with all of these solutions, none of them is super pretty but they're all OK.